src/v/io/CMakeLists.txt

```diff
@@ -1,9 +1,12 @@
+enable_clang_tidy()
+
 v_cc_library(
   NAME io
   SRCS
     persistence.cc
     page.cc
     page_set.cc
+    clang-tidy-helper.cc
```

Comment:
Why is this needed? Is that because otherwise the header's are not checked? Is there a better way?

Reply:
exactly, headers aren't compiled, but cmake models clang-tidy integration as a compiler. we can certainly automate/hide it (partially) using the v_cc_library macro as long as folks list headers.

Reply:
> we can certainly automate/hide it (partially) using the v_cc_library macro as long as folks list headers.

SGTM. I assume that will require us to actually use the `HDRS` field?

Reply:
cache.h for example is included by a .cc test file, but not a file (yet) in a .cc file that is part of the v::io library. tests and normal code use slightly different clang tidy checks, so this helper file just applies the right checks to the right headers, temporarily.